calcom/cal.com
# feat: Sync app credentials between Cal.com & self-hosted platforms

## What does this PR do?



This PR allows self-hosters that have their own app store, to sync their user's 3rd party credentials with their instance of Cal.com. This allows their app store to be the source of truth for apps.

Fixes # (issue)



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?



- Set up the required .env variables that were added in the PR
- Hit the `api/webhook/app-credential` endpoint with the correct webhook secret and encrypted payload
- The new credentials should be created in the DB
- For the affected apps, force a refresh of the tokens. The endpoint they are hitting should simulate what a self-hoster would have made
- Force a refresh of the tokens but this time do not enable app credential syncing

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
